x/auth/vesting/types/vesting_account.go

```diff
@@ -41,56 +41,53 @@ type BaseVestingAccount struct {
 	EndTime          int64     `json:"end_time" yaml:"end_time"`                   // when the coins become unlocked
 }
 
-// NewBaseVestingAccount creates a new BaseVestingAccount object
-func NewBaseVestingAccount(baseAccount *authtypes.BaseAccount, originalVesting sdk.Coins, endTime int64) (*BaseVestingAccount, error) {
-	if (baseAccount.Coins.IsZero() && !originalVesting.IsZero()) || originalVesting.IsAnyGT(baseAccount.Coins) {
-		return &BaseVestingAccount{}, errors.New("vesting amount cannot be greater than total amount")
-	}
+// NewBaseVestingAccount creates a new BaseVestingAccount object. It is the
+// callers responsibility to ensure the base account has sufficient funds with
+// regards to the original vesting amount.
+func NewBaseVestingAccount(baseAccount *authtypes.BaseAccount, originalVesting sdk.Coins, endTime int64) *BaseVestingAccount {
 	return &BaseVestingAccount{
 		BaseAccount:      baseAccount,
 		OriginalVesting:  originalVesting,
 		DelegatedFree:    sdk.NewCoins(),
 		DelegatedVesting: sdk.NewCoins(),
 		EndTime:          endTime,
-	}, nil
+	}
 }
 
-// SpendableCoinsVestingAccount returns all the spendable coins for a vesting account given a
-// set of vesting coins.
+// LockedCoinsFromVesting returns all the coins that are not spendable (i.e. locked)
+// for a vesting account given the current vesting coins and account balance,
+// where the balance only includes tokens of vesting denomination.
```

Comment:
Well, truth is, you can pass anything you want here. It is the caller's responsibility to make sure `balance` is the account's balance of vesting denominations only.

Do you have any suggestions? I don't think I can compare the two sets because it might be valid for one set to have a zero amount.

Reply:
Resolving per the discussion in not even needing balances.